### DISCUSSION TITLE
[call  to  fail] 'models'  parameter  within  the  'messages'

### DISCUSSION THREAD

**Discussion by bahuzh** [open-webui/open-webui#2423] [General]
**Open WebUI Version: 0.1.125**

"The  request  I  sent  includes  an  additional  'models'  parameter  within  the  'messages'  which  has  caused  the  call  to  fail.  This  is  inconsistent  with  the  method  of  invocation  described  in  the  [Chat  API  Reference](https://platform.openai.com/docs/api-reference/chat).

Here  is  the  request:

```json
{
    "chat":  {
      "id":  "",
      "title":  "123",
      "models":  [
        "xxx"
      ],
      "options":  {},
      "messages":  [
        {
          "id":  "d1ec4766-f313-4996-9f77-a172f754a612",
          "parentId":  null,
          "childrenIds":  [
            "7b731959-7fbb-405e-bc5c-58e61bac7308"
          ],
          "role":  "user",
          "content":  "123",
          "models":  [
            "xxx"
          ],
          "timestamp":  1716194703
        },
        {
          "parentId":  "d1ec4766-f313-4996-9f77-a172f754a612",
          "id":  "7b731959-7fbb-405e-bc5c-58e61bac7308",
          "childrenIds":  [],
          "role":  "assistant",
          "content":  "[prompt  illegal]",
          "model":  "xxx",
          "userContext":  null,
          "timestamp":  1716194704,
          "done":  true
        }
      ]
    }
}
```

The  inclusion  of  the  'models'  parameter  within  the  'messages'  is  not  consistent  with  the  expected  format,  leading  to  the  failure  of  the  call."

**Comment by tjbck** [comment]
Requests made to OpenAI API endpoints are perfectly compatible with their specs. It looks like you're confusing Open WebUI's internal chat (ie conversation storage) endpoints as OpenAI compatible endpoints.

![image](https://github.com/open-webui/open-webui/assets/25473318/068b98eb-ca98-4473-a341-0a40a1e0b89a)

**Reply by bahuzh** [reply]
thanks！
